lib/actions/field-trip.js

```diff
@@ -0,0 +1,424 @@
+//     drawItinerary(tripPlan.itineraries[0])
+//     return
+//   }
+//
+//   if (groupPlan == null) {
+//     groupPlan = new otp.modules.planner.TripPlan(null, _.extend(tripPlan.queryParams, { groupSize : groupSize }))
+//   }
+//
+//   if (itinWidget == null) createItinerariesWidget()
+//
+//   var itin = tripPlan.itineraries[0]
+//   var capacity = itin.getGroupTripCapacity()
+//
+//   // if this itin shares a vehicle trip with another one already in use, only use the remainingCapacity (as set in checkTripValidity())
+//   if (itinCapacity) capacity = Math.min(capacity, itinCapacity)
+//
+//   groupPlan.addItinerary(itin)
+//
+//   var transitLegs = itin.getTransitLegs()
+//   for (var i = 0; i < transitLegs.length; i++) {
+//     var leg = transitLegs[i]
+//     bannedSegments.push({
+//       tripId : leg.tripId,
+//       fromStopIndex : leg.from.stopIndex,
+//       toStopIndex : leg.to.stopIndex,
+//     })
+//   }
+//
+//   setBannedTrips()
+//
+//   if (currentGroupSize > capacity) {
+//     // group members remain. plan another trip
+//     currentGroupSize -= capacity
+//     itin.groupSize = capacity
+//     //console.log("remaining: "+currentGroupSize)
+//     itinCapacity = null
+//     planTrip()
+//   } else {
+//     // we're done. show the results
+//     itin.groupSize = currentGroupSize
+//     showResults()
+//   }
+// }
+
+/**
+ * Edit teacher (AKA submitter) notes for a field trip request.
+ */
+export function editSubmitterNotes (request, submitterNotes) {
+  return function (dispatch, getState) {
+    const {callTaker, otp} = getState()
+    const {datastoreUrl} = otp.config
+    if (sessionIsInvalid(callTaker.session)) return
+    const {sessionId} = callTaker.session
+    const queryData = new FormData()
+    queryData.append('sessionId', sessionId)
+    queryData.append('notes', submitterNotes)
+    queryData.append('requestId', request.id)
+    return fetch(`${datastoreUrl}/fieldtrip/editSubmitterNotes`,
+      {method: 'POST', body: queryData}
+    )
+      .then(() => dispatch(fetchFieldTripDetails(request.id)))
+      .catch(err => {
+        alert(`Error editing submitter notes: ${JSON.stringify(err)}`)
+      })
+  }
+}
+
+export function saveRequestTrip (request, outbound, groupPlan) {
+  return function (dispatch, getState) {
+    // If plan is not valid, return before persisting trip.
+    const check = checkPlanValidity(request, groupPlan)
+    if (!check.isValid) return alert(check.message)
+    const requestOrder = outbound ? 0 : 1
+    const type = outbound ? 'outbound' : 'inbound'
+    const preExistingTrip = getTripFromRequest(request, outbound)
+    if (preExistingTrip) {
+      const msg = `This action will overwrite a previously planned ${type} itinerary for this request. Do you wish to continue?`
+      if (!confirm(msg)) return
+    }
+    alert(`TODO: Save trip in request order ${requestOrder}!`)
+    // TODO: Enable saveTrip
+    // dispatch(saveTrip(request, requestOrder))
```

Comment:
Why commented out? I'm not a fan of approving incomplete code.

Reply:
Removed in e8cda1bd131fb1354e32d03268bb94e2ec5583cb